### PR TITLE
Set warn_unused_ignores mypy config to true

### PR DESCRIPTION
**Pull request by kumaranvpl**
# Description

Please include a summary of the change and specify which issue is being addressed. Additionally, provide relevant motivation and context.

Fixes # (issue number)

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples, or any documentation updates)
- [ ] Bug fix (a non-breaking change that resolves an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would disrupt existing functionality)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code adheres to the style guidelines of this project (`scripts/lint.sh` shows no errors)
- [ ] I have conducted a self-review of my own code
- [ ] I have made the necessary changes to the documentation
- [ ] My changes do not generate any new warnings
- [ ] I have added tests to validate the effectiveness of my fix or the functionality of my new feature
- [ ] Both new and existing unit tests pass successfully on my local environment by running `scripts/test-cov.sh`
- [ ] I have ensured that static analysis tests are passing by running `scripts/static-analysis.sh`
- [ ] I have included code examples to illustrate the modifications

### CODE REVIEW

**Comment by kumaranvpl** [comment]
@Lancetnik I have discrepancy between my local development mypy run and CI's mypy run - https://github.com/airtai/faststream/actions/runs/10449142213/job/28930867234. 
Is the same happening to you too? Could you please checkout this branch and run `./scripts/static-analysis.sh`?

BTW, I am using python `3.9.19` in my local and CI uses the same too. mypy version is also same too which is version `1.11.1`.

File: .github/workflows/pr_tests.yaml [hunk 1]
```diff
@@ -49,6 +49,29 @@ jobs:
...
+            3.9
+            3.10
+      - name: Set $PY environment variable
+        run: echo "PY=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_ENV
+      - uses: actions/cache@v4
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ hashFiles('.pre-commit-config.yaml') }}
+      - name: Install Dependencies
+        run: pip install .[docs,testing]
```
Comment (Lancetnik) [inline .github/workflows/pr_tests.yaml:72]: Can we use `uv` instead to be consistent?

File: docs/docs/en/confluent/security.md [hunk 2]
```diff
@@ -85,7 +85,7 @@ from faststream.security import SASLPlaintext
 
 security = SASLPlaintext(
     username="admin",
-    password="password",
+    password="password",  # pragma: allowlist secret
```
Comment (Lancetnik) [inline docs/docs/en/confluent/security.md:88]: Can we do not check `.md` files by bandit or make this file includable with cutting off the comment (like others in this section)? I don't like an idea to show such comments in our documentation

**Review by davorrunje — APPROVED** [review]

**Comment by Lancetnik** [comment]
Seems good, but why we have separated `static-analysis` job now? Can we just run pre-commit without changes?

**Comment by kumaranvpl** [comment]
> Seems good, but why we have separated `static-analysis` job now? Can we just run pre-commit without changes?

Do you mean that instead of running separate `static-analysis` and `pre-commit` jobs, I should just run the `pre-commit` job alone without skipping the static-analysis step?

**Comment by Lancetnik** [comment]
> > Seems good, but why we have separated `static-analysis` job now? Can we just run pre-commit without changes?
> 
> Do you mean that instead of running separate `static-analysis` and `pre-commit` jobs, I should just run the `pre-commit` job alone without skipping the static-analysis step?

I think so, if it is possible

**Review by Lancetnik — APPROVED** [review]

**Pull request merged**